### PR TITLE
Block serialization

### PR DESCRIPTION
**Pull request by TomMcCabe**
Read/write blocks and block headers. All values transmitted in the network are big-endian.

### CODE REVIEW

**Comment by Christewart** [comment]
This is submitted to the wrong branch on my personal repo - it needs to be submitted to 'blockchain_package_and_base_traits' branch

File: src/main/scala/org/bitcoins/core/serializers/blockchain/RawBlockHeaderSerializer.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  /**
+    * Serializes the BlockHeader to a hexadecimal string
+    *
+    * @param blockHeader the block header to be serialized
+    * @return the hexadecimal string representing the block header
+    */
+  def write(blockHeader: BlockHeader) : String = {
+    val headerVersion = BitcoinSUtil.flipEndianess(blockHeader.version.toHexString)
```
Comment (Christewart) [inline src/main/scala/org/bitcoins/core/serializers/blockchain/RawBlockHeaderSerializer.scala:56]: I think there could possibly be a bug here with `blockHeader.version.toHexString`. For instance if we had 
val version = 1 
version.toHexString // gives us "1" NOT "01"
1 byte should always be converted to two characters, NOT 1 character. 

We should replace this line with `BitcoinSUtil.encodeHex(BitcoinSUtil.toByteSeq(blockHeader.version).reverse)`

File: src/main/scala/org/bitcoins/core/serializers/blockchain/RawBlockHeaderSerializer.scala [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    val versionSubPadding = addPrecedingZero(headerVersion)
+    val version = addPadding(8,versionSubPadding)
+
+    val prevHash = blockHeader.previousBlockHash.hex
+    val merkleRoot = blockHeader.merkleRootHash.hex
+
+    val time = BitcoinSUtil.flipEndianess(blockHeader.time.toHexString)
```
Comment (Christewart) [inline src/main/scala/org/bitcoins/core/serializers/blockchain/RawBlockHeaderSerializer.scala:63]: Same as comment above

File: src/main/scala/org/bitcoins/core/serializers/blockchain/RawBlockHeaderSerializer.scala [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    val nBits = BitcoinSUtil.flipEndianess(blockHeader.nBits.toHexString)
```
Comment (Christewart) [inline src/main/scala/org/bitcoins/core/serializers/blockchain/RawBlockHeaderSerializer.scala:64]: Same as comment above

File: src/main/scala/org/bitcoins/core/serializers/blockchain/RawBlockHeaderSerializer.scala [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    val nonce = BitcoinSUtil.flipEndianess(blockHeader.nonce.toHexString)
```
Comment (Christewart) [inline src/main/scala/org/bitcoins/core/serializers/blockchain/RawBlockHeaderSerializer.scala:65]: Same as comment above

File: src/main/scala/org/bitcoins/core/serializers/blockchain/RawBlockSerializer.scala [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  /**
+    * Takes a list of bytes and converts it into a Block
+    *
+    * @param bytes the bytes to be converted to a block
+    * @return the block object parsed from the list of bytes
+    */
+  def read(bytes : List[Byte]) : Block = {
+    val blockHeader : BlockHeader = RawBlockHeaderSerializer.read(bytes.slice(0,80))
+    val txCount : CompactSizeUInt = BitcoinSUtil.parseCompactSizeUInt(bytes.slice(80, bytes.length))
+    val txBytes : Seq[Byte] = bytes.slice((bytes.slice(0,80).size + txCount.size).toInt, bytes.size) //correct. returns byte seq of raw tx.
```
Comment (Christewart) [inline src/main/scala/org/bitcoins/core/serializers/blockchain/RawBlockSerializer.scala]: `bytes.slice(0,80).size` is just equivalent to number 80

File: src/main/scala/org/bitcoins/core/serializers/blockchain/RawBlockSerializer.scala [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    val (transactions, remainingBytes) = parseBlockTransactions(txCount, txBytes)
+      Block(blockHeader, txCount, transactions)
```
Comment (Christewart) [inline src/main/scala/org/bitcoins/core/serializers/blockchain/RawBlockSerializer.scala]: Remove tab here so it is flush with other lines in function definition

**Comment by TomMcCabe** [comment]
Wrong branch.

**Pull request closed without merging**